core/templates/dev/head/profile/PreferencesSpec.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Preferences Controller', function() {
+  describe('PreferencesCtrl', function() {
+    var scope, ctrl, $Http;
+
+    beforeEach(function() {
+      module('oppia');
+    });
+
+    beforeEach(inject(function(_$httpBackend_, $http, $rootScope, $controller) {
+      _$httpBackend_.expectGET('/signuphandler/data').respond({
```

Comment:
Ohh I see. I was fundamentally wrong about it. I thought that backend is still running when we run frontend tests. So I see now. Actually _$httpBackend_() registers a backend url. So when scope queries for url, we return this expectGET() result. Is that right ? We also have to perform $httpBackend.flush() for actual activity of transfer to take place.

We are also changing backend property so I guess I will also need some expectPOST() method too.

Thanx for the info. I will fix the tests and add a new version soon.